### PR TITLE
TEP-0148: Summary/Aggregation API for Tekton Results

### PR DESCRIPTION
**Pull request by avinal**
# Summary

This TEP proposes a new API endpoint for Tekton results that will allow users to
get a summary or aggregation data given a set of filters. This will be useful
for getting a quick overview without having to go through all the response and also
reduce the amount of data that needs to be transferred. This will also provide
a way to get aggregated data for a set of filters.

Co-authored-by: Khurram Baig <kbaig@redhat.com>
Signed-off-by: Avinal Kumar <avinal@redhat.com>

/kind tep

### CODE REVIEW

**Review by khrm — COMMENTED** [review]
/assign @sayan-biswas @enarha

File: teps/0148-summaryaggregation-api-for-tekton-results.md
Comment (khrm) [inline teps/0148-summaryaggregation-api-for-tekton-results.md:400]: We can use `google.protobuf.Struct` here.

File: teps/0148-summaryaggregation-api-for-tekton-results.md
Comment (khrm) [reply teps/0148-summaryaggregation-api-for-tekton-results.md:396]: Also, instead of map<string, Aggregations>, we can have an array.

Comment (avinal) [reply teps/0148-summaryaggregation-api-for-tekton-results.md:396]: I don't think array would work because, then we won't be able to use group name.

Comment (avinal) [reply teps/0148-summaryaggregation-api-for-tekton-results.md:396]: ok understood, changed

**Review by adambkaplan — APPROVED** [review]
/lgtm

As a proposed TEP, this makes a lot of sense to me. Aggregation endpoints make a lot of sense for Results, where data will be stored for longer periods of time.

I propose gating "implementable" on adding additional non-goals related to aggregation features. Which types of queries will this feature/TEP _not_ support? At which point will we recommend community adopters implement their own data extraction/transform systems?

**Comment by avinal** [comment]
> I propose gating "implementable" on adding additional non-goals related to aggregation features. Which types of queries will this feature/TEP not support? At which point will we recommend community adopters implement their own data extraction/transform systems?

I will update the TEP to elaborate non-goals.

**Review by vdemeester — APPROVED** [review]
/hold
cc @wlynch @afrittoli

**Comment by khrm** [comment]
@alan-ghelardi Can you approve this TEP pr?

**Comment by vdemeester** [comment]
/lgtm cancel
/hold cancel

(@adambkaplan see [here](https://github.com/tektoncd/community/blob/main/process/tep-process.md#approval-requirements) why I am "cancelling" the lgtm)

**Comment by adambkaplan** [comment]
/assign @alan-ghelardi 

/assign @dibyom 

Assigning to non-Red Hat Results maintainers.

**Review by dibyom — APPROVED** [review]
/approve

This seems useful. I think we should track the performance metrics to ensure we do not end up transferring the client side load on to the API server

**Pull request merged**
